presto-hive/src/test/java/com/facebook/presto/hive/AbstractTestHiveClient.java

```diff
@@ -1373,29 +1371,29 @@ protected void doTestMismatchSchemaTable(
             MaterializedResult result = readTable(transaction, tableHandle, columnHandles, session, TupleDomain.all(), OptionalInt.empty(), Optional.empty());
             assertEqualsIgnoreOrder(result.getMaterializedRows(), dataAfter.getMaterializedRows());
 
-            int filterCount = afterFilters.size();
-            for (int i = 0; i < filterCount; i++) {
-                RowExpression predicate = afterFilters.get(i);
-                ConnectorTableLayoutHandle layoutHandle = metadata.pushdownFilter(session, tableHandle, predicate, Optional.empty()).getLayout().getHandle();
-
-                // Read all columns with a filter
-                MaterializedResult filteredResult = readTable(transaction, tableHandle, layoutHandle, columnHandles, session, OptionalInt.empty(), Optional.empty());
-
-                Predicate<MaterializedRow> rowPredicate = afterResultPredicates.get(i);
-                List<MaterializedRow> expectedRows = dataAfter.getMaterializedRows().stream().filter(rowPredicate::apply).collect(toList());
-
-                assertEqualsIgnoreOrder(filteredResult.getMaterializedRows(), expectedRows);
-
-                // Read all columns except the ones used in the filter
-                Set<String> filterColumnNames = extractUnique(predicate).stream().map(VariableReferenceExpression::getName).collect(toImmutableSet());
-
-                List<ColumnHandle> nonFilterColumns = columnHandles.stream()
-                        .filter(column -> !filterColumnNames.contains(((HiveColumnHandle) column).getName()))
-                        .collect(toList());
-
-                int resultCount = readTable(transaction, tableHandle, layoutHandle, nonFilterColumns, session, OptionalInt.empty(), Optional.empty()).getRowCount();
-                assertEquals(resultCount, expectedRows.size());
-            }
+//            int filterCount = afterFilters.size();
```

Comment:
@sachdevs Why is this code commented out?

Reply:
We can no longer test by calling metadata.pushdownFilter in our tests. These commented out blocks (see above comments in the PR) were left in just so that it is clear that it is clear that they are obsolete.

It is no longer possible to get the connector table layout handle without invoking the entire plan optimizer and maybe doing an assertPlan(...). I am removing these now.

I'm unsure if there's another way to rewrite this logic or the tests that inherit from AbstractTestHiveClient that depend on this logic. Let me know if you have any alternative ideas.

Reply:
@sachdevs These tests are providing coverage for important scenarios, e.g. schema evolution. I don't believe we have that coverage anywhere else. Simply removing this coverage is undesirable. We need to figure out a replacement.

CC: @bhhari @yingsu00 @oerling

Reply:
@sachdevs @mbasmanova, a quick workaround is to keep the original `pushdownFilter` for `HiveMetadata` (i.e., moving `pushdownFilter` from `HiveFilterPushdown` to `HiveMetadata`). Then in the test, we downcast `ConnectorMetadata` to `HiveMetadata` to still call `pushdownFilter`. WDYT?

Reply:
Hmm but then wouldn't the test be testing a code path that doesn't execute? I.e. nothing actually will call that function but the test.

Reply:
It tests function `pushdownFilter`, isn't it?

Reply:
I can try that out - just need to verify that has no impact on Prism (but should be fine). @mbasmanova ?

Reply:
> It tests function `pushdownFilter`, isn't it?

Yeah, I misunderstood your comment earlier.